### PR TITLE
Migrate IPTools.getHost to async/await

### PR DESCRIPTION
**Pull request by xfix**

### CODE REVIEW

File: server/ip-tools.ts [hunk 1]
```diff
@@ -535,30 +506,32 @@ export const IPTools = new class {
...
-		});
-		socket.on('error', () => {});
-		socket.on('timeout', () => socket.destroy());
-		socket.on('close', () => {
-			if (!connected) {
-				this.connectionTestCache.set(ip, false);
-				callback(false);
-			}
+		return new Promise(resolve => {
+			const socket = require('net').createConnection({
```
Comment (scheibo) [inline server/ip-tools.ts]: import this at the top?

File: server/ip-tools.ts [hunk 2]
```diff
@@ -354,158 +356,128 @@ export const IPTools = new class {
...
+			return 'anchorfree.proxy-nohost';
+		}
+		if (ip === '127.0.0.1') {
+			return 'localhost';
+		}
+		let hosts;
+		try {
+			hosts = await util.promisify(dns.reverse)(ip);
+		} catch (err) {
+			return '' + ip.split('.').slice(0, 2).join('.') + '.unknown-nohost';
```
Comment (Zarel) [inline server/ip-tools.ts]:
```suggestion
			hosts = null;
```

File: server/ip-tools.ts [hunk 3]
```diff
@@ -354,158 +356,128 @@ export const IPTools = new class {
...
+			ip.startsWith('85.237.197.') || ip.startsWith('85.237.222.')
+		) {
+			return 'anchorfree.proxy-nohost';
+		}
+		if (ip === '127.0.0.1') {
+			return 'localhost';
+		}
+		let hosts;
+		try {
+			hosts = await util.promisify(dns.reverse)(ip);
```
Comment (Zarel) [inline server/ip-tools.ts]: I realllllllllly hate `util.promisify`...

It should at least not be used like this.

Comment (Zarel) [reply server/ip-tools.ts]:
```suggestion
			hosts = new Promise(resolve => {
				dns.reverse(ip, (err, hosts) => resolve(hosts));
			});
```

**Comment by Zarel** [comment]
I guess this was closed because Annika's more recent refactor did this? Sorry we didn't get to this if so.

**Pull request closed without merging**